lib/instance.js

```diff
@@ -753,6 +763,15 @@ Instance.prototype.reload = function(options) {
     }
   })
   .then(function(reload) {
+    // persist virtual fields that have been previously set
+    this.Model._virtualAttributes.forEach(function (key) {
+      if (this.dataValues[key] !== undefined) {
+        reload.dataValues[key] = this.dataValues[key];
+      }
```

Comment:
If possible i prefer you use `get(key, {raw: true})` and `set(key, {raw: true})`